### PR TITLE
Lookup table

### PR DESCRIPTION
**Pull request by bminortx**
Adds to README, and improves test for PlannerLib simulation. 
EDIT: It also adds MATLAB capabilities for both LocalSim and the Path Planner.
EDIT: It also implements node calls to add GLObjects to the scene while LocalSim is running.

### CODE REVIEW

**Comment by chachi** [comment]
You committed at least one emacs backup file, the one that starts with #

**Comment by chachi** [comment]
And the .# ones, too.

**Comment by bminortx** [comment]
Ah, that's annoying.

**Comment by chachi** [comment]
Yeah, that usually comes from someone doing `git add .`

File: Applications/PathPlanner/flycheck-SimPlanner.cpp [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
 
 /////////////////////////////////////////////
-// We've replaced what would be a call to a PbMsgs with
-// a call to the heightmap_data_ member variable.
-/////////TODO: Finish editing InitMesh, InitGoals, and SampleTrajectory
-pb::BVP_policy PlannerLibTest::StartPolicy(){
-  InitMesh();
-  InitGoals();
+
+pb::BVP_policy SimPlanner::StartPolicy(pb::BVP_params params){
```
Comment (chachi) [inline Applications/PathPlanner/flycheck-SimPlanner.cpp]: Should pass in policy as by `const pb::BVP_params& params`.

**Comment by chachi** [comment]
Lots of changes, lots of comments :smiley: Take whatever suggestions you'd like.

### Formatting comments
- `{` always on opening line and use whitespace between `)` and `{`.
- Whitespace around operators like `+`, `-`, `*`, `/`, `==`, `<<` (log statements), `<` (inside for loops), etc.
- Whitespace following `,`
- No whitespace following opening `(` or preceding closing `)`, e.g. in argument lists.
- No whitespace following `<` in template argument lists.
- Do we need the `//////////////` lines?
- I would go through and, in Emacs, highlight entire files (`C-x h` and hit tab. That should fix all the indentations.
- There's a lot of multi-line whitespaces that can be condensed.

### C++
- Prefer pass-by-const-reference to pass-by-value. It avoids copying the object if you don't have to modify it.
- Prefer passing in a pointer to an object to modify rather than returning an object from a function. So rather than returning a policy, you'd pass `pb::PolicyMsg* policy` and a function would modify the pointed-to object.
- Lots of raw pointers in this stuff which of course works if you're careful but can be harder to modify w/o problems than using `std::shared_ptr` or `std::unique_ptr`.
- Want to turn the Makefiles into CMake scripts? That way you won't have to hardcode in all your paths, etc. That could be done after this PR, though.
- You include the "pb.cc" files which is not desirable. Would it work if you just add them to the compilation step?

**Comment by bminortx** [comment]
The makefiles actually can't be modified into CMake scripts, due to use of the MEX compiler for MATLAB. I tried this, and lost about four days with no success. I would appreciate it if you could help me out with the .pb.cc thing, too. I'm sure it's an easy thing I just missed.

Everything else is good; I'll see what I can switch up.

**Comment by chachi** [comment]
As far as cosmetics go, this looks good to me. I'll fix that Node related build problem soon, too. Go ahead and merge

**Comment by bminortx** [comment]
Yay! Yeah, the lookup table could use some improvement, but at least it works, haha.

**Pull request merged**